event_question_descr_block/models/event_question_answer.py

```diff
@@ -0,0 +1,37 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+class EventQuestionAnswer(models.Model):
+    _inherit = 'event.question.answer'
+
+    is_blocking = fields.Boolean(string="Is Blocking?")
+
+
+class EventRegistrationAnswer(models.Model):
+    _inherit = 'event.registration.answer'
+
+
+
+    def check_answer_blocking(self, answer_id):
+        answer = self.env['event.question.answer'].browse(answer_id)
+        if answer:
+            if not answer.is_blocking:
+                raise UserError(_(" Based on your Answers you cannot register for this event."))
```

Comment:
This can be combined into a single if: if answer and not answer.is_blocking

Reply:
+1 - Nested ìf` (and `for`) statements are a code smells .

Reply:
@diogocsc  @dreispt  I also had the same thought. But the design mentioned : On "Question Answers" 

    add "Is Blocking?", "When set, this answer is required in order to allow the participant to register. For example, to ask for consent on some required conditions to participate."  It says when Is blocking is set this answer is required in order to allow participants to register.  Should I change the design also because it makes more sense to block if is blocking is set and not other way around?

I will try to not have nested if and for conditions where we can have a work around. Thank you for the review and having patience. Am glad to get feedback in detail.

Reply:
@tirth-osi from what I see there is no need to change the design at https://pm.opensourceintegrators.com/web#id=29500&menu_id=176&cids=1&action=1098&model=project.task&view_type=form.
Though I would like to understand why the last part of it is strikethrough